File: dayan-server/dayan-modules/dayan-module-finance/src/main/java/com/dayan/finance/entity/FinanceAccount.java
package com.dayan.finance.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dayan.common.mybatis.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
/**
 * 表 finance_account 对应实体。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("finance_account")
public class FinanceAccount extends BaseEntity {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;


    /** 账目编码 */
    private String accountCode;

    /** 账目方向 */
    private Integer direction;

    /** 对象类型 */
    private String accountType;

    /** 对象编码 */
    private String targetCode;

    /** 对象名称 */
    private String targetName;

    /** 业务类型 */
    private String bizType;

    /** 业务编码 */
    private String bizCode;

    /** 应收/应付总额 */
    private BigDecimal totalAmount;

    /** 已收/已付金额 */
    private BigDecimal receivedAmount;

    /** 未收/未付金额 */
    private BigDecimal remainAmount;

    /** 到期日期 */
    private LocalDate dueDate;

    /** 最近收款/付款时间 */
    private LocalDateTime lastReceiveTime;

    /** 状态 */
    private Integer accountStatus;

    /** 备注 */
    private String remark;
}

File: dayan-admin/src/types/finance-account.ts
/**
 * 应收应付账目（finance_account）相关类型。
 *
 * 字段对齐后端 com.dayan.finance 包下：
 * - FinanceAccountVO / FinanceAccountQueryDTO / CreateAccountDTO / AccountReceiveDTO。
 *
 * 类型约定：Integer→number，BigDecimal→number，LocalDate/LocalDateTime→string。
 */
import type { PageQuery } from '@/types/common'

// ==================== 账目方向 ====================

/** 账目方向：1=应收 / 2=应付 */
export enum AccountDirection {
  /** 应收 */
  RECEIVABLE = 1,
  /** 应付 */
  PAYABLE = 2
}

export const ACCOUNT_DIRECTION_OPTIONS = [
  { label: '应收', value: AccountDirection.RECEIVABLE },
  { label: '应付', value: AccountDirection.PAYABLE }
] as const

// ==================== 账目状态 ====================

/**
 * 账目状态：0=待收付 / 1=部分收付 / 2=已结清 / 3=已逾期 / 4=已坏账
 */
export enum AccountStatus {
  /** 待收付 */
  PENDING = 0,
  /** 部分收付 */
  PARTIAL = 1,
  /** 已结清 */
  SETTLED = 2,
  /** 已逾期 */
  OVERDUE = 3,
  /** 已坏账 */
  BAD_DEBT = 4
}

export const ACCOUNT_STATUS_OPTIONS = [
  { label: '待收付', value: AccountStatus.PENDING },
  { label: '部分收付', value: AccountStatus.PARTIAL },
  { label: '已结清', value: AccountStatus.SETTLED },
  { label: '已逾期', value: AccountStatus.OVERDUE },
  { label: '已坏账', value: AccountStatus.BAD_DEBT }
] as const

/**
 * 应收应付账目实体（后端 FinanceAccountVO）。
 */
export interface FinanceAccount {
  id?: number
  /** 账目编号（系统生成） */
  accountCode?: string
  /** 账目方向：1=应收/2=应付 */
  direction?: AccountDirection
  /** 对象类型：channel/supplier/agent */
  accountType?: string
  /** 对象编码 */
  targetCode?: string
  /** 对象名称 */
  targetName?: string
  /** 业务类型：equity_purchase/scene_fee/service_fee */
  bizType?: string
  /** 业务编码 */
  bizCode?: string
  /** 应收/应付总额 */
  totalAmount?: number
  /** 已收/付金额 */
  receivedAmount?: number
  /** 剩余应收/应付金额 */
  remainAmount?: number
  /** 到期日期 */
  dueDate?: string
  /** 最近一次收/付款时间 */
  lastReceiveTime?: string
  /** 状态 */
  accountStatus?: AccountStatus
  remark?: string
  createdAt?: string
}

/**
 * 应收应付账目分页查询参数（后端 FinanceAccountQueryDTO）。
 */
export interface FinanceAccountQuery extends PageQuery {
  accountCode?: string
  direction?: AccountDirection
  accountType?: string
  targetCode?: string
  bizType?: string
  bizCode?: string
  accountStatus?: AccountStatus
  /** 到期日期 ≤ */
  dueDateTo?: string
}
